src/core/Akka.Remote/Transport/Streaming/StreamingTcpTransport.cs

```diff
@@ -0,0 +1,384 @@
+                ));
+        }
+
+        public sealed override bool Write(IO.ByteString payload)
+        {
+            return _queue.OfferAsync(payload)
+                .Result is QueueOfferResult
+                .Enqueued;
+        }
+        
+
+        
+
+
+        public override void Disassociate()
+        {
+            _queue.Complete();
+        }
+
+        public void Notify(IHandleEvent inboundPayload)
+        {
+            _listener.Notify(inboundPayload);
+        }
+    }
+
+    class StreamingTcpTransport : Transport
+    {
+        public sealed override string SchemeIdentifier { get; protected set; } =
+            "tcp";
+
+        private StreamingTcpTransportSettings TransportSettings;
+        private ActorMaterializer _mat;
+        //private DotNettyTransportSettings Settings;
+
+        private Source<Tcp.IncomingConnection, Task<Tcp.ServerBinding>>
+            _connectionSource;
+
+        protected readonly TaskCompletionSource<IAssociationEventListener>
+            AssociationListenerPromise;
+
+        private Task<Tcp.ServerBinding> _serverBindingTask;
+        private Address _addr;
+        private EndPoint _listenAddress;
+        private EndPoint _outAddr;
+        private TaskCompletionSource<Address> _boundAddressSource;
+
+        public StreamingTcpTransport(ActorSystem system, Config config)
+        {
+            AssociationListenerPromise =
+                new TaskCompletionSource<IAssociationEventListener>();
+            System = system;
+            _mat = ActorMaterializer.Create(System,
+                ActorMaterializerSettings.Create(System).WithDispatcher("akka.remote.default-remote-dispatcher"),
+                namePrefix: "streaming-transport");
+            if (system.Settings.Config.HasPath("akka.remote.dot-netty.tcp"))
+            {
+                var dotNettyFallbackConfig =
+                    system.Settings.Config.GetConfig(
+                        "akka.remote.dot-netty.tcp");
+                config = dotNettyFallbackConfig.WithFallback(config);
+            }
+
+            if (system.Settings.Config.HasPath("akka.remote.helios.tcp"))
+            {
+                var heliosFallbackConfig =
+                    system.Settings.Config.GetConfig("akka.remote.helios.tcp");
+                config = heliosFallbackConfig.WithFallback(config);
+            }
+
+            TransportSettings = StreamingTcpTransportSettings.Create(config);
+
+            SocketOptions = ImmutableList.Create<Inet.SocketOption>(
+                new Inet.SO.ReceiveBufferSize(
+                    TransportSettings.SocketReceiveBufferSize),
+                new Inet.SO.SendBufferSize(
+                    TransportSettings.SocketSendBufferSize),
+                new Inet.SO.ByteBufferPoolSize(TransportSettings.TransportReceiveBufferSize),
+                new Inet.SO.WorkerDispatcher("akka.remote.default-remote-dispatcher"));
```

Comment:
Should the IO.Tcp Actors be running on Remote Dispatcher? or on their own (perhaps same pool as materializer?)

Reply:
Yes they should IMHO - I'd have them share the remote dispatcher